workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/WorkflowDeployer.java

```diff
@@ -61,16 +64,25 @@ public void addWorkflow(Path workflowFile) throws IOException, ProcessingExcepti
     log.debug("Adding a new workflow");
     Workflow workflow = SwadlParser.fromYaml(workflowFile.toFile());
     BpmnModelInstance instance = workflowEngine.parseAndValidate(workflow);
-    Pair<String, Boolean> deployedWorkflow = deployedWorkflows.get(workflowFile);
+
+    //TODO: Given that {id, version} is unique, do we still get workflow by file? it would require an index to make it correctly fast
```

Comment:
the function id would be the combined fields id, and version, a unique index on these two fields is needed.